students/jjakub/session07/html_render/html_render.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+"""
+Class: Python 210 B, Au2018
+Exercise: Session 06, Mailroom Part 04
+Student: Jason Jakubiak
+
+A class-based system for rendering html.
+"""
+
+# This is the framework for the base class
+class Element(object):
+    """
+    Abstract base class for rendering an Html element
+    """
+    tag = "html"
+    indent = "    "
+
+    def __init__(self, content=None, **kwargs):
+        self.attributes = kwargs
+        self.contents = []
+        if content is not None:
+            self.contents = [content]
+
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+
+    def attrubutes(self):
+        attrb = " ".join([' {}="{}"'.format(key, val) 
```

Comment:
Since you already added a leading blank space in `' {}="{}"'`, it'd be better if you just do `.join()` on an empty string (`""`), not a string with one blank space like now (`" "`).

Reply:
That makes sense. I've updated the code and will commit with the final version.